DSharpPlus/Cache/CacheKeys.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace DSharpPlus.Cache;
+
+using Entities;
+
+public interface ICacheKey
+{
+    public const string KeyPrefix = "dsharpplus";
+}
+
+public readonly record struct GuildCacheKey : ICacheKey
```

Comment:
for all these structs - why not use a primary constructor? the `readonly` constraint on the struct should enforce init-only properties anyways, even though record structs are settable

also, personally i'm a fan of breaking them into their own files, but oh well